praw/reddit.py

```diff
@@ -579,10 +607,41 @@ def post(
             None).
 
         """
-        data = self.request(
-            "POST", path, data=data or {}, files=files, params=params
-        )
-        return self._objector.objectify(data)
+        retries = 0
+        while True:
+            _raise = True
+            try:
+                return self._objectify_request(
+                    "POST", path, data=data or {}, files=files, params=params
+                )
+            except RedditAPIException as exception:
+                print(exception)
+                for item in exception.items:
```

Comment:
Please extract this logic into a private helper method. The call could look something like this:


```
except RedditAPIException as exception:
    rate_limit_seconds = _extract_rate_limit_seconds(exception)
    if rate_limit_seconds is not None:
        time.sleep(rate_limit_seconds)
        return self._objectify_request(...)
    raise
```